dbt/adapters/snowflake/relation.py

```diff
@@ -33,3 +51,48 @@ def is_dynamic_table(self) -> bool:
     @classproperty
     def DynamicTable(cls) -> str:
         return str(SnowflakeRelationType.DynamicTable)
+
+    @classproperty
+    def get_relation_type(cls) -> Type[SnowflakeRelationType]:
+        return SnowflakeRelationType
+
+    @classmethod
+    def from_runtime_config(cls, runtime_config: RuntimeConfigObject) -> RelationConfigBase:
+        model_node: ModelNode = runtime_config.model
+        relation_type = SnowflakeRelationType(model_node.config.materialized)
+
+        if relation_config := cls.relation_configs.get(relation_type):
+            return relation_config.from_model_node(model_node)
+
+        raise DbtRuntimeError(
+            f"from_runtime_config() is not supported for the provided relation type: {relation_type}"
+        )
+
+    @classmethod
+    def dynamic_table_config_changeset(
+        cls,
+        dynamic_table: SnowflakeDynamicTableConfig,
+        relation_results: RelationResults,
+    ) -> Optional[SnowflakeDynamicTableConfigChangeset]:
+        config_changeset = SnowflakeDynamicTableConfigChangeset()
+
+        existing_dynamic_table = SnowflakeDynamicTableConfig.from_relation_results(
+            relation_results
+        )
+        assert isinstance(existing_dynamic_table, SnowflakeDynamicTableConfig)
+
+        if dynamic_table.target_lag != existing_dynamic_table.target_lag:
+            config_changeset.target_lag = SnowflakeDynamicTableTargetLagConfigChange(
+                action=RelationConfigChangeAction.alter,
+                context=dynamic_table.target_lag,
+            )
+
+        if dynamic_table.warehouse != existing_dynamic_table.warehouse:
+            config_changeset.warehouse = SnowflakeDynamicTableWarehouseConfigChange(
+                action=RelationConfigChangeAction.alter,
+                context=dynamic_table.warehouse,
+            )
+
+        if config_changeset.has_changes:
+            return config_changeset
+        return None
```

Comment:
nit: put an `else` ;p

Reply:
::gets on soap box and raises pinky:: My understanding is that it's more idiomatic to return the `None` outside of the `else` block since the `else` block isn't doing anything. That's how I've generally seen it, so the `else: return None` construct looks strange to me, as if we could get past it somehow and not return anything (because it's indented). A compromise could be:
```python
return config_changeset if config_changeset.has_changes else None
```

Reply:
Bruh I trust you but wow. Conventions be something. I'm happy with the compromise hehe